.buildkite/pipeline.yml

```diff
@@ -0,0 +1,22 @@
+steps:
+  - command: |
+      . /var/lib/buildkite-agent/.nix-profile/etc/profile.d/nix.sh
+      echo "build:ci --host_platform=@rules_haskell//haskell/platforms:linux_x86_64_nixpkgs" > .bazelrc.local
+      nix-shell --arg docTools false --pure --run \
+          'bazel build --config ci //tests:run-tests'
+      # TODO(Profpatsch) re-add a nixpkgs startup script
+      # and enable this test again
+      nix-shell --arg docTools false --pure --run \
+          './bazel-ci-bin/tests/run-tests --skip "/startup script/"'
+      nix-shell --arg docTools false --pure --run \
+          'bazel coverage //tests/... --config ci --build_tag_filters "coverage-compatible" --test_tag_filters "coverage-compatible" --test_output=all'
+    label: "Run tests (Nixpkgs)"
+    timeout: 30
+  - command: |
+      echo "common:ci --build_tag_filters -requires_hackage,-requires_lz4,-requires_zlib,-requires_doctest,-requires_c2hs,-requires_threaded_rts,-dont_test_with_bindist" > .bazelrc.local
+      # XXX: See .bazelrc [backward compatible options] for the the rational behind this flag
+      echo "build --incompatible_use_python_toolchains=false" >> .bazelrc.local
+      bazel build --config ci //tests/...
+      ./tests/run-start-script.sh
```

Comment:
While tinkering with sourcing the nix profile inside the agent hooks, I noticed this step cannot run when the nix profile is sourced before, due to `ld.gold` not finding `gmp`. Is this a known issue / expected?

https://buildkite.com/tweag-1/rules-haskell/builds/29#b7b15f5e-178c-4bc4-b891-6e2bc2332cb7/21-174

Reply:
Yes, I noticed this too. I'm not sure why this is. But it seems fair to keep Nix and non-nix environments separate.

Reply:
Can you open an issue about this, so we can follow-up on it later?

Reply:
so there's already https://github.com/tweag/rules_haskell/issues/704 about that, which we closed as WONTFIX. We might want to revisit that, assuming the CI agents run on NixOS, and we want to test the bindist feature.